### ISSUE TITLE
Alternative implementation of Data::Frame, and more

### ISSUE THREAD

**Issue by stphnlyd** [EntropyOrg/p5-Data-Frame#27]
@zmughal 

I am not able to reach you via your CPAN mail address, so hereby I am creating this issue to discuss this. 

I am working a port of R ggplot2 to Perl. My repo is at https://github.com/stphnlyd/perl5-Chart-GGPlot. It's still quite incomplete compared to ggplot2, but I've basically done the core of it and have got quite a few things working. You can find the examples at https://github.com/stphnlyd/perl5-Chart-GGPlot/tree/master/examples

Data::Frame is a key dependency of my work (you know the ggplot idea requires some data frame implementaion). And I've made quite a lot updates upon your Data::Frame. Now part of the changes are at https://github.com/stphnlyd/p5-Data-Frame/tree/alt-pdlsv (for my implementation of PDL::SV and PDL::Factor), and more at at https://github.com/stphnlyd/perl5-Chart-GGPlot/tree/master/lib/Data/Frame (now as Data::Frame::More but I hope to have it to be just Data::Frame.)

Basically I am seeking to be the co-maintainer of Data::Frame on CPAN, hoping to get all my changes into Data::Frame. And on Github I would like to join @EntropyOrg to work on your repo. Of course I will try my best to keep back-compatibility with what you have now, although there will have to be some incompatibilities (like bumping Perl requirement from 5.12 to 5.16) Would like to hear your ideas. 

Thanks!

**Comment by stphnlyd** [comment]
@zmughal 
I ended up releasing https://metacpan.org/pod/Alt::Data::Frame::ButMore to cpan. Contact me if you would like it to be merged into your project.

**Comment by zmughal** [comment]
Hi, sorry for leaving this open for so long.

@mohawk2 just mentioned this to me again and I have given both of you comaint on the modules in the dist:

```
$ comaint Data::Frame SLOYD
$ comaint Data::Frame ETJ
```

You can go ahead and incorporate whatever you need into a new release. You should have write access to this repository. Let me know if everything works.

**Comment by stphnlyd** [comment]
Thanks. I will get some time in the next couple of weeks, to review my
implementation first, and see how we can merge changes to Data::Frame.

Regards,
Stephan


Zaki Mughal [sivoais] ***@***.***> 于 2021年7月27日周二 03:19写道：

**Comment by stphnlyd** [comment]
covered by #29

**Comment by zmughal** [comment]
Great! Thanks @stphnlyd.

Another point before doing the next release: could you add co-maint of the new namespaces to ETJ (@mohawk2)  and ZMUGHAL (me)? That way any of us can do a release.

- <https://metacpan.org/dist/Data-Frame/permissions>
- <https://metacpan.org/dist/Alt-Data-Frame-ButMore/permissions>

**Comment by mohawk2** [comment]
I have a further suggestion: harmonise each distro so all its packages have the same owner, whoever that is. Given what *I believe* we're aiming for here, I'd suggest that the new packages should be first-come-d to @zmughal, with co-maint for me and @stphnlyd. Then a new release of the Alt distro with only the Alt module in it, so it becomes vestigial.
